### PR TITLE
feat: User table SQL queries

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This change introduces the SQL queries required for the GoCert authorization flow as defined in TE091. The password salting and hashing algorithm required for storage is also done at the database level.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ui/src/app/types.ts
Comment (saltiyazan) [inline ui/src/app/types.ts:1]: Not like it is super important, but why did you change the names here?

**Review by saltiyazan — APPROVED** [review]

File: ui/src/app/types.ts
Comment (kayra1) [reply ui/src/app/types.ts:1]: When I first wrote the handler I wasn't sure how to map the struct to different json key names. I know now, so I changed it to be consistent with all the other json objects we have.
